### DISCUSSION TITLE
KF5SyntaxHighlighting_DIR-NOTFOUND

### DISCUSSION THREAD

**Discussion by buiphukhuyen** [cpeditor/cpeditor#1238] [Q&A]
How to fix this?
<img width="1186" alt="Screenshot 2023-10-21 at 15 50 14" src="https://github.com/cpeditor/cpeditor/assets/43597300/fa485693-072b-459c-9a8a-cf3eb93c4662">

**Comment by ouuan** [comment]
Have you cloned the git submodules?

**Comment by buiphukhuyen** [comment]
Sure, I using git: git clone --recurse-submodules https://github.com/cpeditor/cpeditor.git
<img width="396" alt="Screenshot 2023-10-21 at 16 04 04" src="https://github.com/cpeditor/cpeditor/assets/43597300/19132f3e-0b58-4e9e-b59c-fc8b34198e34">

**Comment by ouuan** [comment]
You may refer to this: https://github.com/cpeditor/cpeditor/blob/0022c1c8376a6cc011f6e335255d02a824c8ebbb/.github/workflows/build_test.yml#L58-L77

Or maybe you can install it via system package manager? I'm not familiar with macOS.

**Reply by buiphukhuyen** [reply]
I tried using system package manager on MacOS and Windows both had this error.

<img width="994" alt="Screenshot 2023-10-21 at 16 14 48" src="https://github.com/cpeditor/cpeditor/assets/43597300/35deb5f2-25b3-4faa-839f-495ff27eb373">

**Comment by buiphukhuyen** [comment]
I checkout the v7 branch and there is sucess, but the master branch has the above error. Please check the project again!
<img width="1055" alt="Screenshot 2023-10-21 at 16 31 34" src="https://github.com/cpeditor/cpeditor/assets/43597300/fa6fe2a3-c7a0-4ef8-b137-804924c29135">

**Reply by ouuan** [reply]
The v7 branch is using the old QCodeEditor so of course you won't meet errors with KSyntaxHighlighting.

**Reply by ouuan** [reply]
And a surprise is that even v6.10 is more up-to-date than the v7.0 branch.

**Reply by buiphukhuyen** [reply]
Haha, When I running v7 
<img width="906" alt="Screenshot 2023-10-21 at 17 42 06" src="https://github.com/cpeditor/cpeditor/assets/43597300/14f0ce00-ff25-44cd-a988-2c45eead2241">

**Reply by buiphukhuyen** [reply]
In addition to the library error "KF5SyntaxHighlighting_DIR-NOTFOUND" when build with CMake, there is also "ECM (Extra-Cmake-Modules".
